src/io/observe.cc

```diff
@@ -152,10 +156,11 @@ void ObservePoint::find_observe_point(Mesh &mesh) {
     for (unsigned int i_candidate=0; i_candidate<process_list.size(); ++i_candidate) {
         unsigned int i_elm=process_list[i_candidate];
         Element & elm = mesh.element[i_elm];
-        arma::mat map = elm.element_map();
+        arma::mat map;
+        arma::vec projection;
```

Comment:
Make case according to dimension here and move rest of calculation on single element into auxiliary class template.
(insluding ObservePoint::snap method)